### PR TITLE
[SYCL][NFC] Fix Static Analyzer bugs

### PR DESCRIPTION
**Pull request by smanna12**
This patch fixes two possible null pointer dereferences issues:

    1. In CodeGenModule::Release(): Dereferencing a pointer that might be
       nullptr RD->getAttr() when calling getAspectsMD().

    2. In Sema::ActOnFinishFunctionBody(): Passing null pointer FD to
       hasAttr, which dereferences it.

Signed-off-by: Soumi Manna <soumi.manna@intel.com>

### CODE REVIEW

File: clang/lib/Sema/SemaDecl.cpp [hunk 1]
```diff
@@ -16555,7 +16555,7 @@ Decl *Sema::ActOnFinishFunctionBody(Decl *dcl, Stmt *Body,
     checkTypeSupport(FD->getType(), FD->getLocation(), FD);
 
   // Handle free functions.
-  if (LangOpts.SYCLIsDevice && FD->hasAttr<SYCLDeviceAttr>() && Body &&
+  if (FD && LangOpts.SYCLIsDevice && FD->hasAttr<SYCLDeviceAttr>() && Body &&
```
Comment (premanandrao) [inline clang/lib/Sema/SemaDecl.cpp:16558]: Suggestion:

Add:
``  if (!FD)
      return dcl;
``
before 16545, and remove the FD checks on lines 16545, 16554 and the new 16558.

Comment (smanna12) [reply clang/lib/Sema/SemaDecl.cpp]: Thanks @premanandrao for the reviews and suggestion.

Done

**Review by premanandrao — APPROVED** [review]

**Comment by smanna12** [comment]
Thank you @premanandrao for reviews!

**Comment by smanna12** [comment]
@intel/llvm-gatekeepers , This PR is ready to merge. Thank you

**Pull request merged**